hivemind/client/averaging/allreduce.py

```diff
@@ -182,7 +187,11 @@ async def run(self) -> Sequence[torch.Tensor]:
     async def accumulate_part_streaming(self, source: Endpoint, stream_messages: Iterable[runtime_pb2.Tensor]
                                         ) -> Iterable[runtime_pb2.Tensor]:
         """ accumulate_part using streams of serialized tensors. Used to prevent duplicate work in serialization """
-        tensor_part: torch.Tensor = deserialize_torch_tensor(combine_from_streaming(stream_messages))
+        try:
+            tensor_part: torch.Tensor = deserialize_torch_tensor(combine_from_streaming(stream_messages))
+        except RuntimeError as e:
+            raise AllreduceException(f"Could not deserialize tensor part from {source} for streaming {e}")
```

Comment:
Ideally, we should narrow down the scope of this exception:
1) Do we want to catch errors that happen only because of streaming? If so, enclose `combine_from_streaming` inside the try-except block
2) Do we want to catch any kind of specifically deserialization errors? If so, move this check inside `deserialize_torch_tensor`. If you still want an AllreduceException here, you can then catch a specific SerializerException

Reply:
The problem is that errors in combine_from_streaming only manifest in deserialize_torch_tensor .